### PR TITLE
Add Task Model

### PR DESCRIPTION
**Pull request by indiephi**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ android {
     buildToolsVersion "25.0.3"
     defaultConfig {
         applicationId "com.kyphipraniti.fitnesstasks"
-        minSdkVersion 16
+        minSdkVersion 21
```
Comment (pranitishettiwar) [inline app/build.gradle:8]: why are we changing minSdkVersion ? lot of people are still on kitkat.

File: .idea/misc.xml [hunk 2]
```diff
@@ -37,7 +43,7 @@
     <ConfirmationsSetting value="0" id="Add" />
     <ConfirmationsSetting value="0" id="Remove" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="JDK" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="JDK" project-jdk-type="JavaSDK">
```
Comment (pranitishettiwar) [inline .idea/misc.xml]: Do we need all these idea file changes ? they usually show up dirty, but you need to add them to gitignore

**Review by pranitishettiwar — COMMENTED** [review]
few comments

File: app/build.gradle (hunk 1 above)
Comment (statham) [reply app/build.gradle:8]: agreed

File: app/src/main/java/com/kyphipraniti/fitnesstasks/model/Task.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            this.deadline = iso8601Format.parse(datetime);
+            this.amount = cursor.getLong(cursor.getColumnIndex("amount"));
+            this.completed = (cursor.getInt(cursor.getColumnIndex("completed")) != 0);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+    }
+
+    Task(long id, String deadline, long amount, boolean completed) {
+        Date parsedDeadline = null;
```
Comment (statham) [inline app/src/main/java/com/kyphipraniti/fitnesstasks/model/Task.java]: do we want to allow deadlines to be null?

File: app/src/main/res/layout-v21/fragment_calendar.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+             xmlns:tools="http://schemas.android.com/tools"
+             android:layout_width="match_parent"
+             android:layout_height="match_parent"
+             tools:context=".fragments.CalendarFragment">
+
+    <DatePicker
```
Comment (statham) [inline app/src/main/res/layout-v21/fragment_calendar.xml]: you should probably give this an id

File: app/src/main/res/layout/fragment_calendar.xml [hunk 5]
```diff
@@ -1,13 +1,19 @@
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (statham) [inline app/src/main/res/layout/fragment_calendar.xml:1]: i'm a bit confused why you have this fragment layout twice?

**Review by statham — COMMENTED** [review]
some questions but overall good. Would it be feasible for you to add firebase database support to this as well? if it adds too much work load then we can just save it for another pr

File: app/build.gradle (hunk 1 above)
Comment (Vsin) [reply app/build.gradle:8]: We'll have to add separate layout files for datePickerMode attribute in layout/fragment_calendar to support 16.

File: app/src/main/res/layout/fragment_calendar.xml (hunk 5 above)
Comment (Vsin) [reply app/src/main/res/layout/fragment_calendar.xml:1]: removed second layout.

**Pull request merged**